catppuccin-sddm: init unstable-2022-12-03

Chaostheorie opened the pull request:
###### Description of changes

Adds the catppuccin sddm theme. I might need some guidance on where the package should be located. For now it is located as a QT package however that can be changed.

This derivation has been runnning on my setup for the better part of 6 months and I wanted to upstream it. I'm not quite sure how the runtime dependencies on QT quickcontrols 2 and sddm components should be integrated as they appear to be part of sddm anyways.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] ~~x86_64-darwin~~ (sddm not applicable)
  - [ ] ~~aarch64-darwin~~ (sddm not applicable)
- [ ] ~~For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix /manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))~~ (sddm not applicable)
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Chaostheorie committed:
feat: Add catppuccin sddm theme

Chaostheorie committed:
Merge branch 'master' of github.com:Chaostheorie/nixpkgs

SuperSandro2000 commented on pkgs/top-level/qt5-packages.nix:
```diff
@@ -235,6 +235,7 @@ in (kdeFrameworks // plasmaMobileGear // plasma5 // plasma5.thirdParty // kdeGea
   # Not a library, but we do want it to be built for every qt version there
   # is, to allow users to choose the right build if needed.
   sddm = callPackage ../applications/display-managers/sddm { };
+  sddm-catppuccin = callPackage ../applications/display-managers/sddm/sddm-catppuccin.nix {};
```
In qt packages only offficial qt things or libraries are meant to be put in. The theme should just live in top-level.

Chaostheorie replied:
it has been moved to the top-level and renamed to fit in with the other catppuccin themes.

SuperSandro2000 commented on pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:
```diff
@@ -0,0 +1,27 @@
+  meta = {
+    description = "Soothing pastel theme for SDDM";
+    homepage = "https://github.com/catppuccin/sddm";
+    maintainers = [ ];
```
Maintainer must be set.

Chaostheorie replied:
resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

SuperSandro2000 commented on pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:
```diff
@@ -0,0 +1,27 @@
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "sddm";
+    # Main @ bde6932
```
```suggestion
```

Chaostheorie replied:
resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

SuperSandro2000 commented on pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:
```diff
@@ -0,0 +1,27 @@
+    mv src/catppuccin-${flavor} $out/share/sddm/themes/catppuccin-${flavor}
+  '';
+
+  src = fetchFromGitHub {
```
src should be ordered after version

Chaostheorie replied:
resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

SuperSandro2000 commented on pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:
```diff
@@ -0,0 +1,27 @@
+  pname = "sddm-catppuccin-theme";
+  version = "0.0.0b0";
+
+  dontBuild = true;
```
Is this actually required? Is stdenv detecting some makefile?

Chaostheorie replied:
this wasn't required. It must've been an artifact from another theme.

SuperSandro2000 commented on pkgs/applications/display-managers/sddm/sddm-catppuccin.nix:
```diff
@@ -0,0 +1,27 @@
+{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:
+stdenv.mkDerivation {
+  pname = "sddm-catppuccin-theme";
+  version = "0.0.0b0";
```
```suggestion
{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:

stdenv.mkDerivation {
  pname = "sddm-catppuccin-theme";
  version = "unstable-XXXX-XX-XX";
```
XXXX-XX-XX should be the date of the commit in iso format

Chaostheorie replied:
resolved in [517edfb](https://github.com/NixOS/nixpkgs/pull/240990/commits/517edfbaeaad7c040ffa212334e4584816f2a2d1)

Chaostheorie commented:
Thank you for the quick review, I will fix the suggested changes asap.

Chaostheorie committed:
fix: Move catppuccin-sddm theme and apply suggested changes

The derivation was renamed to fit in with the other catppuccin themes and moved to the top-level.
Additionally, a maintainer was added and the version number changed.

Chaostheorie committed:
fix: sort the maintainers list

Chaostheorie committed:
fix; add github id for cobalt

SuperSandro2000 commented on maintainers/maintainer-list.nix:
````diff
@@ -1,61 +1,61 @@
-    };
-    ```
+  If `github` begins with a numeral, `handle` should be prefixed with an underscore.
+  ```nix
+  _1example = {
+  github = "1example";
+  };
+  ```
 
-    Add PGP/GPG keys only if you actually use them to sign commits and/or mail.
+  Add PGP/GPG keys only if you actually use them to sign commits and/or mail.
 
-    To get the required PGP/GPG values for a key run
-    ```shell
-    gpg --fingerprint <email> | head -n 2
-    ```
+  To get the required PGP/GPG values for a key run
+  ```shell
+  gpg --fingerprint <email> | head -n 2
+  ```
 
-    !!! Note that PGP/GPG values stored here are for informational purposes only, don't use this file as a source of truth.
+  !!! Note that PGP/GPG values stored here are for informational purposes only, don't use this file as a source of truth.
 
-    More fields may be added in the future, however, in order to comply with GDPR this file should stay as minimal as possible.
+  More fields may be added in the future, however, in order to comply with GDPR this file should stay as minimal as possible.
 
-    When editing this file:
-     * keep the list alphabetically sorted, check with:
-         nix-instantiate --eval maintainers/scripts/check-maintainers-sorted.nix
-     * test the validity of the format with:
-         nix-build lib/tests/maintainers.nix
+  When editing this file:
+  * keep the list alphabetically sorted, check with:
+  nix-instantiate --eval maintainers/scripts/check-maintainers-sorted.nix
+  * test the validity of the format with:
+  nix-build lib/tests/maintainers.nix
 
-    See `./scripts/check-maintainer-github-handles.sh` for an example on how to work with this data.
+  See `./scripts/check-maintainer-github-handles.sh` for an example on how to work with this data.
````
Please remove those formatting changes

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -18297,7 +18308,7 @@
     email = "me@zzzsy.top";
     github = "zzzsyyy";
-    githubId = 	59917878;
+    githubId = 59917878;
```
we can keep this

SuperSandro2000 commented on maintainers/maintainer-list.nix:
```diff
@@ -3127,6 +3127,17 @@
   };
+  cobalt = {
+    name = "Cobalt";
+
+    email = "cobalt@cobalt.rocks";
+    github = "Chaostheorie";
+    githubId = 42151227;
+
```
```suggestion
    email = "cobalt@cobalt.rocks";
    github = "Chaostheorie";
    githubId = 42151227;
```

Chaostheorie committed:
Update maintainers/maintainer-list.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

keenanweaver commented on pkgs/data/themes/catppuccin-sddm/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ stdenv, lib, fetchFromGitHub, flavor ? "mocha" }:
+stdenv.mkDerivation {
+  pname = "sddm-catppuccin-theme";
+  version = "unstable-2022-12-03";
```
```suggestion
  version = "unstable-2023-08-08";
```

keenanweaver commented on pkgs/data/themes/catppuccin-sddm/default.nix:
```diff
@@ -0,0 +1,25 @@
+  src = fetchFromGitHub {
+    owner = "catppuccin";
+    repo = "sddm";
+    rev = "bde6932e1ae0f8fdda76eff5c81ea8d3b7d653c0";
+    sha256 = "ceaK/I5lhFz6c+UafQyQVJIzzPxjmsscBgj8130D4dE=";
```
```suggestion
    rev = "7fc67d1027cdb7f4d833c5d23a8c34a0029b0661";
    hash = "sha256-SjYwyUvvx/ageqVH5MmYmHNRKNvvnF3DYMJ/f2/L+Go=";
```
Bumped to latest

Chaostheorie commented:
Closed in favor of #255808
